back/src/forms/repository/form/update.ts

```diff
@@ -0,0 +1,38 @@
+import { Form, Prisma } from "@prisma/client";
+import { GraphQLContext } from "../../../types";
+import { indexForm } from "../../elastic";
+import { LogMetadata, RepositoryFnDeps } from "../types";
+import buildFindFullFormById from "./findFullFormById";
+
+export type UpdateFormFn = (
+  where: Prisma.FormWhereUniqueInput,
+  data: Prisma.FormUpdateInput,
+  logMetadata?: LogMetadata
+) => Promise<Form>;
+
+const buildUpdateForm: (deps: RepositoryFnDeps) => UpdateFormFn =
+  deps => async (where, data, logMetadata) => {
+    const { user, prisma } = deps;
+
+    const updatedForm = await prisma.form.update({
+      where,
+      data
+    });
+
+    await prisma.event.create({
+      data: {
+        streamId: updatedForm.id,
+        actor: user.id,
+        type: "BsddUpdated",
+        data: { content: data } as Prisma.InputJsonObject,
+        metadata: { ...logMetadata, authType: user.auth }
+      }
+    });
+
+    const fullForm = await buildFindFullFormById(deps)(updatedForm.id);
+    await indexForm(fullForm, { user } as GraphQLContext);
```

Comment:
Le seul truc "dommage", c'est que du coup l'indexation rentre dans les transactions. Mais c'est inévitable avec ce fonctionnement.
A terme, on pourrait peut être gérer l'indexation dans une queue pour ne pas avoir à attendre le retour. Et trouver un moyen différent de mettre à jour le front (p-e des server-side events avec un front qui écoute sur un endpoint correspondant à son siret et se maj dès que le back lui envoie un event)

Reply:
oui, la job queue est là, c'est très simple de le faire

Reply:
Est-ce que c'est pas déjà un peu ce que fait Elastic Search par défaut, quand `refresh` n'est pas définit à `wait_for` ? C'est pas une queue d'évènement mais comme je le comprends Elastic Search a sa propre gestion pour répondre vite, quitte à retarder la mise à jour d'un index.

Si c'est toujours pas assez rapide, une alternative pourrait être de ne pas `await` la réponse. Ça veut dire qu'un échec n'influerait pas la transaction mais c'est vrai aussi avec une queue d'évènement. Par contre une queue peut gérer le retry (pas sûr que ça soit le cas actuellement avec les queues qu'on a ?).

Reply:
Le cas d'un appel serveur est en effet pas très grave, c'est plus les appels front en `wait_for`.
Et oui la queue permettrait de gérer le retry, c'est le gros avantage que j'y vois.